### ISSUE TITLE
Finish question if course staff leaves queue

### ISSUE THREAD

**Issue by nwalters512** [illinois/queue#104]
If a member of course staff is still "answering" a question when they leave the queue, the question should be marked as answered. Alternatively, prevent course staff from leaving the queue until they "finish" their question.

**Comment by JParisFerrer** [comment]
How will the second option work with being removed from active duty by another user? Is it not the case that only the answerer can mark the question as answered? Or is it just UI only? Another thing is that some people just leave and never leave the on duty list which wouldn't be solved by either of these (except that questions can still be deleted by the asker I guess?)

**Comment by nwalters512** [comment]
Ah true, I forgot that people can be taken off the queue by others. So it's probably just best to just mark it as answered. Or maybe we don't want to mark it as answered, and leave it on the queue but without a staff answering it? Not sure what the best option is here.

**Comment by JParisFerrer** [comment]
It's a shame because I prefer the second option logically. We probably don't want to just unmark it, because then it sits there at the top of the queue?

**Comment by nwalters512** [comment]
Yeah... the second one makes more sense, but it requires people to always leave the queue, which they clearly don't always do.

**Comment by jackieo5023** [comment]
Reviving this thread... How about prompting whoever removes someone who's on duty with something along the lines of "Are you sure you want to remove this user/leave the queue? The question they are answering will be finished."
